jablib/src/main/java/org/jabref/logic/importer/fetcher/EuropePmcFetcher.java

```diff
@@ -0,0 +1,148 @@
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+import org.jabref.logic.importer.IdBasedParserFetcher;
+import org.jabref.logic.importer.ParseException;
+import org.jabref.logic.importer.Parser;
+import org.jabref.logic.importer.util.JsonReader;
+import org.jabref.model.entry.Author;
+import org.jabref.model.entry.AuthorList;
+import org.jabref.model.entry.BibEntry;
+import org.jabref.model.entry.Month;
+import org.jabref.model.entry.field.StandardField;
+import org.jabref.model.entry.field.UnknownField;
+import org.jabref.model.entry.types.EntryType;
+import org.jabref.model.entry.types.StandardEntryType;
+
+import kong.unirest.core.json.JSONArray;
+import kong.unirest.core.json.JSONException;
+import kong.unirest.core.json.JSONObject;
+
+public class EuropePmcFetcher implements IdBasedParserFetcher {
+
+    @Override
+    public URL getUrlForIdentifier(String identifier) throws URISyntaxException, MalformedURLException {
+        return URI.create("https://www.ebi.ac.uk/europepmc/webservices/rest/search?query=" + identifier + "&resultType=core&format=json").toURL();
+    }
+
+    @Override
+    public Parser getParser() {
+        return inputStream -> {
+            JSONObject response = JsonReader.toJsonObject(inputStream);
+            if (response.isEmpty()) {
+                return List.of();
+            }
+            return List.of(jsonItemToBibEntry(response));
+        };
+    }
+
+    private BibEntry jsonItemToBibEntry(JSONObject item) throws ParseException {
+        try {
+            JSONObject result = item.getJSONObject("resultList").getJSONArray("result").getJSONObject(0);
+
+            System.out.println(result.toString(2));
+
+            EntryType entryType = StandardEntryType.Article;
+            if (result.has("pubTypeList")) {
+                for (Object o : result.getJSONObject("pubTypeList").getJSONArray("pubType")) {
+                    if ("letter".equalsIgnoreCase(o.toString())) {
+                        entryType = StandardEntryType.Article;
+                        break;
+                        // TODO: handle other types e.g. books
+                    }
+                }
+            }
+
+            BibEntry entry = new BibEntry(entryType);
+
+            entry.setField(StandardField.TITLE, result.optString("title"));
+            entry.setField(StandardField.ABSTRACT, result.optString("abstractText"));
+
+            entry.setField(StandardField.YEAR, result.optString("pubYear"));
+            entry.setField(StandardField.VOLUME, result.optString("journalVolume"));
+            entry.setField(StandardField.ISSUE, result.optString("journalIssue"));
+
+            // Format pages with double dash
+            String pages = result.optString("pageInfo");
+            if (pages != null && !pages.isEmpty()) {
+                pages = pages.replace("-", "--");
+                entry.setField(StandardField.PAGES, pages);
+            }
```

Comment:
Can we use org.jabref.logic.formatter.bibtexfields.NormalizePagesFormatter here?

Reply:
done in postcleanup